File: app/src/main/java/com/example/retrofit_pokemon/retrofit2/PokemonAPI.java
package com.example.retrofit_pokemon.retrofit2;

import android.util.JsonReader;

import com.google.gson.JsonObject;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface PokemonAPI {

    @GET("pokemon/id{id}")
    Call<JsonObject> getPokemonbyId (@Path("id")String id);
}

File: app/src/main/java/com/example/retrofit_pokemon/retrofit2/PokemonAPI.kt
package com.example.retrofit_pokemon.retrofit2

import com.google.gson.JsonObject
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface PokemonAPI {
    @GET("pokemon/id{id}")
    fun getPokemonById(@Path("id") id: String?): Call<List<JsonObject?>?>?
}